### PR TITLE
refactor(loadbalance): change lb next method return Endpoint with ownership

### PR DESCRIPTION
**Pull request by wbtlb**
### What is changed and how it works?

What's Changed:
- change lb `next` method return value with ownership

### CODE REVIEW

File: pisa-proxy/proxy/loadbalance/src/balance.rs [hunk 1]
```diff
@@ -97,3 +96,36 @@ impl Balance {
...
+            addr: String::from("127.0.0.1:3307"),
+        };
+        balance.add(ep1.clone());
+        balance.add(ep2.clone());
+        for _i in 0..10 {
+            println!("{:#?}", balance.next());
+        }
+        
+    }
+}
```
Comment (dongzl) [inline pisa-proxy/proxy/loadbalance/src/balance.rs:131]: suggest add blank line at end of file.

File: pisa-proxy/proxy/loadbalance/src/balance.rs [hunk 2]
```diff
@@ -97,3 +96,36 @@ impl Balance {
...
+        };
+        let ep2 = Endpoint {
+            weight: 1,
+            name: String::from("dasheng002"),
+            db: String::from("db002"),
+            user: String::from("root"),
+            password: String::from("root"),
+            addr: String::from("127.0.0.1:3307"),
+        };
+        balance.add(ep1.clone());
```
Comment (xuanyuan300) [inline pisa-proxy/proxy/loadbalance/src/balance.rs:124]: Remove `clone` here.

File: pisa-proxy/proxy/loadbalance/src/balance.rs [hunk 3]
```diff
@@ -97,3 +96,36 @@ impl Balance {
...
+        balance.add(ep2.clone());
```
Comment (xuanyuan300) [inline pisa-proxy/proxy/loadbalance/src/balance.rs:125]: Remove `clone` here.

File: pisa-proxy/proxy/loadbalance/src/balance.rs [hunk 4]
```diff
@@ -97,3 +96,36 @@ impl Balance {
...
+        for _i in 0..10 {
+            println!("{:#?}", balance.next());
```
Comment (xuanyuan300) [inline pisa-proxy/proxy/loadbalance/src/balance.rs:127]: Please add `assert` to check `next`.

File: pisa-proxy/proxy/loadbalance/src/balance.rs (hunk 1 above)
Comment (wbtlb) [reply pisa-proxy/proxy/loadbalance/src/balance.rs:131]: I will check

**Review by xuanyuan300 — APPROVED** [review]

**Pull request merged**